src/Arcus.Messaging.ServiceBus.Core/Extensions/AzureClientFactoryBuilderExtensions.cs

```diff
@@ -0,0 +1,77 @@
+        }
+
+        /// <summary>
+        /// Registers a <see cref="ServiceBusClient" /> instance into the Azure client factory <paramref name="builder"/>
+        /// via a connection string available via the <paramref name="connectionStringSecretName"/> in the Arcus secret store.
+        /// </summary>
+        /// <remarks>
+        ///     Make sure that the Arcus secret store is registered in the application before using this extension (<a href="https://security.arcus-azure.net/features/secret-store">more info</a>)
+        ///     as the Azure Service Bus connection string will be retrieved via the <paramref name="connectionStringSecretName"/>.
+        /// </remarks>
+        /// <param name="builder">The Azure client factory builder to add the Azure Service Bus client.</param>
+        /// <param name="connectionStringSecretName">The secret name that corresponds with the Azure Service Bus connection string that is registered in the Arcus secret store.</param>
+        /// <param name="configureOptions">The function to configure additional user option that alters the behavior of the Azure Service Bus interaction.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="builder"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="connectionStringSecretName"/> is blank.</exception>
+        public static IAzureClientBuilder<ServiceBusClient, ServiceBusClientOptions> AddServiceBusClient(
+            this AzureClientFactoryBuilder builder,
+            string connectionStringSecretName,
+            Action<ServiceBusClientOptions> configureOptions)
+        {
+            Guard.NotNull(builder, nameof(builder), "Requires an Azure client factory builder to add the Azure Service Bus client");
+            Guard.NotNullOrWhitespace(connectionStringSecretName, nameof(connectionStringSecretName), "Requires a non-blank secret name to retrieve the Azure Service Bus connection string from the Arcus secret store");
+            
+            return builder.AddClient<ServiceBusClient, ServiceBusClientOptions>((options, serviceProvider) =>
+            {
+                var secretProvider = serviceProvider.GetService<ISecretProvider>();
+                if (secretProvider is null)
+                {
+                    throw new InvalidOperationException(
+                        "Requires an Arcus secret store registration to retrieve the connection string to authenticate with Azure Service Bus while creating an Service Bus client instance," 
+                        + "please use the 'services.AddSecretStore(...)' or 'host.ConfigureSecretStore(...)' (https://security.arcus-azure.net/features/secret-store)");
+                }
+
+                string connectionString = secretProvider.GetRawSecretAsync(connectionStringSecretName).GetAwaiter().GetResult();
```

Comment:
Still a bit skeptical about this part. We could create 'proxy' classes around the `ServiceBusClient`, `ServiceBusSender` and `ServiceBusReceiver` so that upon any asynchronous call, the connection string secret would be retrieved, but was wondering if that would be worth the effort AND if that would be stable enough to use it if one of these classes changes.

Reply:
I understand, but I do not see an easy way out of this, except maybe providing non-async methods for retrieving the secrets.

Reply:
The idea to do this sync secret retrieval was already in the past mentioned, here: https://github.com/arcus-azure/arcus.security/issues/259, but we didn't really go through with it (yet). I think bc of the effort it would take for the minor benefits.